evm/precompiles/loomprecompiles/testdata/TestLoomNativeApi.sol

```diff
@@ -0,0 +1,82 @@
+pragma solidity >=0.4.21;
+
+contract TestLoomNativeApi {
+    function TestMappedLoomAccount(string memory ethChainId, bytes32 hash, uint8 v, bytes32 r, bytes32 s) public view returns (address) {
+        return MappedAccount(ethChainId, ecrecover(hash, v, r, s));
+    }
+
+    function TestMappedAccount(string memory ethChainId, bytes32 hash, uint8 v, bytes32 r, bytes32 s, string memory toChain) public view returns (address) {
+        return MappedAccount(ethChainId, ecrecover(hash, v, r, s), toChain);
+    }
+
+    // LoomApi library
```

Comment:
Why is this library code duplicated here?

Reply:
Bearing in mind that the point is to test `addressmapping.go` and not the Solidity code, I found it easier to 'statically link' the LoomNativeApi and the test functions in this fashion rather than dynamically link the libraries. Dynamically linking the libraries involves modifying the TestLoomNativeApi.sol binary at runtime with the address of the library contract in the active EVM.

Reply:
I could add some utility code to allow dynamically linking Solidity libraries in unit tests.